webview-ui/src/components/ui/hooks/useSelectedModel.ts

```diff
@@ -218,16 +218,8 @@ function getSelectedModel({
 		}
 		// kilocode_change begin
 		case "kilocode": {
-			const displayModelId = {
-				gemini25: "Gemini 2.5 Pro",
-				gemini25flashpreview: "Gemini 2.5 Flash Preview",
-				claude37: "Claude 3.7 Sonnet",
-				gpt41: "GPT 4.1",
-			}
-			const id = displayModelId[(apiConfiguration?.kilocodeModel as keyof typeof displayModelId) ?? "claude37"]
-
 			// Use the fetched models from routerModels
-			if (routerModels?.["kilocode-openrouter"] && apiConfiguration?.kilocodeModel) {
+			if (routerModels["kilocode-openrouter"] && apiConfiguration.kilocodeModel) {
```

Comment:
I don't think `routerModels` and `apiConfiguration` can be undefined: https://github.com/Kilo-Org/kilocode/blob/f55e517a269d3f8205983de65204264b2d6757f5/webview-ui/src/components/ui/hooks/useSelectedModel.ts#L59

Reply:
I think it's unlikely `apiConfiguration.kilocodeModel` was undefined when I got the crash, so that leaves `routerModels["kilocode-openrouter"]`... maybe that's because of network flakiness?

Reply:
So i get the other change, but this makes a crash more probable, right?

Reply:
but the change is basically provably correct isn't it?

I think the excessive checking is undesirable here, because the next time somebody looks at this code the first question they will ask is whether routerModels and apiConfiguration can be undefined.